### DISCUSSION TITLE
How to replace useReducer with Zustand

### DISCUSSION THREAD

**Discussion by Nemmcy** [pmndrs/zustand#954] [Q&A]
To prevent some unavoidable and useless re-renders that come with Contexts, I decided to try Zustand. It's quite easy to replace Contexts with simple states, but not so easy for more complex ones that use useReducer.

Here is a [CodeSandbox](https://codesandbox.io/s/firebase-context-api-78tfwg) of a simplified version of my Firebase Context that uses useReducer to dispatch the user and authentication state. I would love to share how I tried to setup this in a Zustand store, but honestly it is useless.

Here are the only examples I found where useReducer was replaced with Zustand: [example1](https://codesandbox.io/s/vf10u?file=/src/Chords.js), [example2](https://codesandbox.io/s/u7303?file=/src/z/store.js).

But I just don't see how to do it in my case.... so if you are willing to take on your time to try to refactor my implementation with Zustand, you are more than welcome!

**Comment by dai-shi** [comment]
I'm not sure which part you got lost.
You don't need to use context in your case with zustand.
To sync with firebase auth, you can do it at the module level instead of useEffect.

I hope you can break down your problems and ask a smaller question.
